lib/matplotlib/legend.py

```diff
@@ -507,10 +512,16 @@ def __init__(self, parent, handles, labels,
         self._loc_used_default = tmp  # ignore changes done by _set_loc
 
         # figure out title fontsize:
-        if title_fontsize is None:
-            title_fontsize = mpl.rcParams['legend.title_fontsize']
-        tprop = FontProperties(size=title_fontsize)
-        self.set_title(title, prop=tprop)
+        self.title_prop = FontProperties._from_any(title_prop)
```

Comment:
```suggestion
        title_prop = FontProperties._from_any(title_prop)
```

Reply:
For the sake of L516 (it checks the original value of 'title_prop' argument, I add a new local variable 'title_prop_fp' here.